tests/lib/testing_classes.py

```diff
@@ -100,13 +102,13 @@ def download_test_data(self, test_data_folder, persist, request):
                 if ext and ext.lstrip('.') in handler_class.IMPLEMENTED_ZIP_FORMATS:  # noqa: E501
                     handler_class.unzip(os.path.join(tmpdir, file_name))
 
-                yield tmpdir
+            yield tmpdir
 
-                persist = (persist or self.PERSIST or
-                           self.is_test_failed(request))
-                if not persist:
-                    print("Removing {}".format(tmpdir))
-                    shutil.rmtree(tmpdir)
+            persist = (persist or self.PERSIST or
+                       self.is_test_failed(request) or dump_databases)
+            if not persist:
+                print("Removing {}".format(tmpdir))
+                shutil.rmtree(tmpdir)
```

Comment:
Moved to tear down code outside of the for loop, cause we dont need to tear down per test file?

Reply:
Yes, no need to tear down output file structure after each test.